### ISSUE TITLE
Flutter Engine w/Dart tip buildbot is broken.

### ISSUE THREAD

**Issue by aam** [dart-lang/sdk#30689]
[flutter-engine-linux](https://uberchromegw.corp.google.com/i/client.dart.internal/builders/flutter-engine-linux) shows that it has been failing for a while now.
Error (below) seems to indicate invalid(outdated?) analyzer on the dart bot. Running the same on local machine works fine.

```
Analyzing frontend_server...
Unhandled exception:
UnimplementedError
#0      CompilationUnitElementForLink.resolveTypeRef (package:analyzer/src/summary/link.dart:1096)
#1      CompilationUnitElementForLink.resolveTypeRef.getTypeArgument (package:analyzer/src/summary/link.dart:1103)
#2      new List.generate (dart:core/list.dart:142)
#3      ClassElementForLink_Class.buildType (package:analyzer/src/summary/link.dart:647)
#4      CompilationUnitElementForLink.resolveTypeRef (package:analyzer/src/summary/link.dart:1113)
#5      VariableElementForLink.declaredType (package:analyzer/src/summary/link.dart:5196)
#6      VariableElementForLink.type (package:analyzer/src/summary/link.dart:5287)
#7      PropertyAccessorElementForLink_Variable.returnType (package:analyzer/src/summary/link.dart:4548)
#8      PropertyAccessorElementForLink_Variable.asStaticType (package:analyzer/src/summary/link.dart:4501)
#9      ExprTypeComputer._doPushReference (package:analyzer/src/summary/link.dart:2485)
#10     ExprTypeComputer.compute (package:analyzer/src/summary/link.dart:2073)
#11     TypeInferenceNode.evaluate (package:analyzer/src/summary/link.dart:4957)
#12     TypeInferenceDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:4808)
#13     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:147)
#14     DependencyWalker.walk (package:front_end/src/dependency_walker.dart:166)
#15     VariableElementForLink.inferredType (package:analyzer/src/summary/link.dart:5226)
#16     FieldElementForLink_ClassField.link (package:analyzer/src/summary/link.dart:2700)
#17     ClassElementForLink_Class.link (package:analyzer/src/summary/link.dart:673)
#18     CompilationUnitElementInBuildUnit.link (package:analyzer/src/summary/link.dart:1308)
#19     LibraryElementInBuildUnit.link (package:analyzer/src/summary/link.dart:3672)
#20     LibraryCycleNode.link (package:analyzer/src/summary/link.dart:3408)
#21     LibraryCycleDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:3330)
#22     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:147)
#23     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#24     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#25     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#26     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#27     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#28     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#29     DependencyWalker.walk.strongConnect (package:front_end/src/dependency_walker.dart:113)
#30     DependencyWalker.walk (package:front_end/src/dependency_walker.dart:166)
#31     LibraryCycleForLink.ensureLinked (package:analyzer/src/summary/link.dart:3373)
#32     Linker.link (package:analyzer/src/summary/link.dart:3897)
#33     relink (package:analyzer/src/summary/link.dart:132)
#34     link (package:analyzer/src/summary/link.dart:113)
#35     new LibraryContext.forSingleLibrary.<anonymous closure>.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:116)
#36     PerformanceLog.run (package:front_end/src/base/performace_logger.dart:34)
#37     new LibraryContext.forSingleLibrary.<anonymous closure> (package:analyzer/src/dart/analysis/library_context.dart:115)
#38     PerformanceLog.run (package:front_end/src/base/performace_logger.dart:34)
#39     new LibraryContext.forSingleLibrary (package:analyzer/src/dart/analysis/library_context.dart:53)
#40     AnalysisDriver._createLibraryContext (package:analyzer/src/dart/analysis/driver.dart:1092)
#41     AnalysisDriver._computeUnitElement (package:analyzer/src/dart/analysis/driver.dart:1054)
#42     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:790)
<asynchronous suspension>
#43     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:1538)
<asynchronous suspension>
#44     AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:1481)
#45     Driver._createAnalysisContext (package:analyzer_cli/src/driver.dart:617)
#46     Driver._analyzeAllImpl (package:analyzer_cli/src/driver.dart:190)
<asynchronous suspension>
#47     Driver._analyzeAll (package:analyzer_cli/src/driver.dart:166)
<asynchronous suspension>
#48     Driver.start (package:analyzer_cli/src/driver.dart:145)
<asynchronous suspension>
#49     main (file:///b/build/slave/dart-sdk-linux-stable/build/sdk/pkg/analyzer_cli/bin/analyzer.dart:11)
#50     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:263)
#51     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)
Failed.
step returned non-zero exit code: 1
@@@STEP_FAILURE@@@

```

Local:
```
╰─➤  PATH=~/projects/dart/sdk/xcodebuild/ReleaseX64/dart-sdk/bin:$PATH flutter/travis/analyze.sh
Analyzing frontend_server...

Analyzing dart:ui library...

╰─➤  
```

@mkustermann @whesse

**Comment by aam** [comment]
@bwilkerson , your https://github.com/dart-lang/sdk/commit/af7bf48e7dd20fcba08c901f96eef7c76d44df14 seem to be the first CL, where this started breaking

**Comment by aam** [comment]
The issue seems to be with 1.24.2 dartanalyzer, that is crashing on analyzing code changed in https://github.com/dart-lang/sdk/commit/af7bf48e7dd20fcba08c901f96eef7c76d44df14. With dartanalyzer 1.25.0-dev.9.0 flutter/travis/analyze.sh completes without issues.

**Comment by whesse** [comment]
Note that we can and will upgrade the checked-in SDK in the dart checkout at tools/sdks/[linux,windows,macos]/dart-sdk as needed.  It is currently at  1.25.0-dev.7.0.  So I don't think this is relevant to your problem.

**Comment by aam** [comment]
This is puzzling then. I was able to reproduce the crash when I used 1.24.2 and it worked fine with 1.25.0-dev.9.0(as well as 7.0 and 11.0).

It would be interesting to confirm that `dartanalyzer --packages=flutter/frontend_server/.packages --options flutter/analysis_options.yaml flutter/frontend_server` on that buildbot fails, and what version of dartanalyzer is that.

**Comment by aam** [comment]
@whesse , can you please help taking a look at the bot? We fixed similar issue with flutter engine travis build by updating dart version on travis buildbot (as part of this PR https://github.com/flutter/engine/commit/1b367b06f1c7cab0be10a7a8ad381ccb7b124acf).

**Comment by mraleph** [comment]
@aam the bot is building successfully now.

**Comment by aam** [comment]
@mraleph cool! what was the root cause I'm curious?

**Comment by mraleph** [comment]
My guess would be you fixed it with 1b367b06f1c7cab0be10a7a8ad381ccb7b124acf

Recipe just calls `api.step('analyze dart_ui', ['/bin/sh', 'flutter/travis/analyze.sh'])`.

Failing builds were all building with 3df8c88f89d6c68aed39aee40105df8c277b5942 and then everything started passing at 1b367b06f1c7cab0be10a7a8ad381ccb7b124acf.
